modules/navier_stokes/src/fv/kernels/NSFVKernel.C

```diff
@@ -0,0 +1,134 @@
+{
+  Moose::FV::interpolate(
+      Moose::FV::InterpMethod::Average, v, elem_v, neighbor_v, *_face_info, true);
+
+  if (m == Moose::FV::InterpMethod::RhieChow)
+  {
+    // Get pressure gradient
+    const VectorValue<ADReal> & grad_p = _p_var->adGradSln(*_face_info);
+
+    // Get uncorrected pressure gradient
+    const VectorValue<ADReal> & unc_grad_p = _p_var->uncorrectedAdGradSln(*_face_info);
+
+    auto tup = Moose::FV::determineElemOneAndTwo(*_face_info, *_p_var);
+    const Elem * const elem_one = std::get<0>(tup);
+    const Elem * const elem_two = std::get<1>(tup);
+    const bool elem_is_elem_one = std::get<2>(tup);
+    mooseAssert(
+        elem_is_elem_one ? elem_one == &_face_info->elem() && elem_two == _face_info->neighborPtr()
+                         : elem_one == _face_info->neighborPtr() && elem_two == &_face_info->elem(),
+        "The determineElemOneAndTwo utility determined the wrong value for elem_is_elem_one");
+
+    const Point & elem_one_centroid =
+        elem_is_elem_one ? _face_info->elemCentroid() : _face_info->neighborCentroid();
+    const Point * const elem_two_centroid =
+        elem_two
+            ? (elem_is_elem_one ? &_face_info->neighborCentroid() : &_face_info->elemCentroid())
+            : nullptr;
+    Real elem_one_volume =
+        elem_is_elem_one ? _face_info->elemVolume() : _face_info->neighborVolume();
+    Real elem_two_volume =
+        elem_two ? (elem_is_elem_one ? _face_info->neighborVolume() : _face_info->elemVolume()) : 0;
+
+    // Now we need to perform the computations of D
+    const ADReal & elem_one_a = rcCoeff(*elem_one);
+    ADReal face_a = elem_one_a;
+
+    mooseAssert(elem_two ? _subproblem.getCoordSystem(elem_one->subdomain_id()) ==
+                               _subproblem.getCoordSystem(elem_two->subdomain_id())
+                         : true,
+                "Coordinate systems must be the same between the two elements");
+
+    Real coord;
+    coordTransformFactor(_subproblem, elem_one->subdomain_id(), elem_one_centroid, coord);
+
+    elem_one_volume *= coord;
+    Real face_volume = elem_one_volume;
+
+    if (elem_two && this->hasBlocks(elem_two->subdomain_id()))
+    {
+      const ADReal & elem_two_a = rcCoeff(*elem_two);
+
+      Moose::FV::interpolate(Moose::FV::InterpMethod::Average,
+                             face_a,
+                             elem_one_a,
+                             elem_two_a,
+                             *_face_info,
+                             elem_is_elem_one);
+
+      coordTransformFactor(_subproblem, elem_two->subdomain_id(), *elem_two_centroid, coord);
+      elem_two_volume *= coord;
+      Moose::FV::interpolate(Moose::FV::InterpMethod::Average,
+                             face_volume,
+                             elem_one_volume,
+                             elem_two_volume,
+                             *_face_info,
+                             elem_is_elem_one);
```

Comment:
is it better to interpolate volume and areas separately then take the ratio or to interpolate D directly?

Reply:
I suppose that I don't know the answer. Equation 15.35 in Moukalleds FVM in CFD has D_face = V_face / a_face, so I suppose that I setup this calculation according to that.

Note that little `a` doesn't denote area. It denotes some linearized term coefficients for the INSFV discretization. They are shown throughout chapter 15 of Moukalled. (I can share that text with you if you want to read it, and you don't have it)

Reply:
I have the book, thanks though. 
15.52 has DUf be half of DUF and DUC so I think D should be interpolated.

Reply:
Nice. Yea that's for the Rhie-Chow interpolation itself, so much better reference than equation 15.35 which is for staggered grid discretizations. I'll push up a change for that soon

Reply:
Fixed in https://github.com/idaholab/moose/pull/15644/commits/0eea50a50215eebe3935e2105768fa7ecfc5633f